packages/apollo-server-integration-testsuite/src/ApolloServer.ts

```diff
@@ -2699,7 +2705,19 @@ export function testApolloServer<AS extends ApolloServerBase>(
         const unsubscribeSpy = jest.fn();
         const { gateway, triggers } = makeGatewayMock({ unsubscribeSpy });
         triggers.resolveLoad({ schema, executor: async () => ({}) });
-        await createApolloServer({ gateway });
+        const server = (await createApolloServer({ gateway })).server;
+        if (options.serverlessFramework) {
+          // Serverless frameworks execute ApolloServer.start() in a dangling
+          // promise in the server constructor. To ensure that the server has
+          // started in the case of serverless, we make a query against it,
+          // which forces us to wait until after start.
+          //
+          // This is also required because ApolloServer.stop() was not designed
```

Comment:
Good point, not great that there's a lack of support for this (given that stop can get called by signal handlers). Want to add an XXX comment to ApolloServer.stop() noting that it might not behave well when called concurrently with start (or before it, even)?

Reply:
Yep, I've added an XXX comment to `ApolloServerBase.stop()` to warn about the behavior.